Applications/DataExplorer/DataView/DiagramView/DetailWindow.cpp

```diff
@@ -124,9 +124,8 @@ void DetailWindow::resizeWindow()
 
 void DetailWindow::addList(DiagramList* list)
 {
-	GeoLib::Color* c = GeoLib::getRandomColor();
-	QColor colour((*c)[0], (*c)[1], (*c)[2]);
-	delete c;
+	GeoLib::Color const c = GeoLib::getRandomColor();
```

Comment:
though I'm not sure if there is any difference, `GeoLib::Color const c(GeoLib::getRandomColor());` may be safer, i.e. avoiding copy.